File: examples/angry_birds/angry_birds_python/brick.py
"""Destructible brick/block — takes velocity-based damage on collision."""

from src.engine.core import GameObject, MonoBehaviour
from src.engine.audio import AudioSource
from src.engine.physics.rigidbody import Rigidbody2D


class Brick(MonoBehaviour):

    def __init__(self):
        super().__init__()
        self.health: float = 70.0
        self.max_health: float = 70.0

    def on_collision_enter_2d(self, collision):
        other_rb = collision.game_object.get_component(Rigidbody2D)
        if other_rb is None:
            return

        damage = collision.relative_velocity.magnitude * 10
        if damage < 5:
            return  # Ignore tiny bumps

        if damage >= 10:
            audio = self.get_component(AudioSource)
            if audio:
                audio.play()

        self.health -= damage

        if self.health <= 0:
            GameObject.destroy(self.game_object)
        else:
            # Darken color based on damage
            self._update_color()

    def _update_color(self):
        from src.engine.rendering.renderer import SpriteRenderer
        sr = self.get_component(SpriteRenderer)
        if sr is None:
            return
        ratio = max(0.3, self.health / self.max_health)
        r, g, b = sr.color
        sr.color = (int(r * ratio), int(g * ratio), int(b * ratio))

File: examples/angry_birds/angry_birds_unity/Brick.cs
using UnityEngine;

namespace AngryBirds
{
    [RequireComponent(typeof(AudioSource))]
    public class Brick : MonoBehaviour
    {
        public float Health = 70f;
        public float MaxHealth = 70f;

        void OnCollisionEnter2D(Collision2D col)
        {
            if (col.gameObject.GetComponent<Rigidbody2D>() == null) return;

            float damage = col.relativeVelocity.magnitude * 10;
            if (damage < 5) return;

            if (damage >= 10)
                GetComponent<AudioSource>().Play();

            Health -= damage;

            if (Health <= 0)
            {
                Destroy(gameObject);
            }
            else
            {
                UpdateColor();
            }
        }

        private void UpdateColor()
        {
            var sr = GetComponent<SpriteRenderer>();
            if (sr == null) return;

            float ratio = Mathf.Max(0.3f, Health / MaxHealth);
            Color c = sr.color;
            sr.color = new Color(c.r * ratio, c.g * ratio, c.b * ratio, c.a);
        }
    }
}
